js/classes/conversation.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * A conversation is an entity, in that it exists in the entities array and updates,
+ * but really it is a container/handler for the dialog bubbles in its array.
+ * Conversations can be stored in entities, but when they are active, the game goes into Dialog mode.
+ */
+class Conversation {
+    /**
+     * @param {Array<DialogBubble>} array The array containing all possible dialog bubbles in this conversation.
+     * @param {boolean} isNew Is it Chad's first time seeing this conversation? Default: true.
+     */
+    constructor(array, isNew = true) {
+        this.array = array;
+        this.curr = 0;
+        this.new = isNew;
+    };
+
+    /**
+     * As the conversation is an entity, it must have an update loop. The update really just
+     * listens for user input, and will change the dialog bubble shown accordingly.
+     */
+    update() {
+        // Is the user wanting to move on to the next dialog bubble?
+        if (GAME.user.continuingConversation) {
+            // The user has pressed space.
+
+            const currentBubble = this.array[this.curr];
+
+            if (currentBubble.currentLine >= currentBubble.targetLines.length) {
```

Comment:
There are 3 instance fields of any DialogBubble (let's call an instance db) you should be familiar with: db.targetLines (contains all text, finished on construction), db.visibleLines (the text that's actually shown), and db.currentLine. db.currentLine tells us which line of text we are currently adding to - if it is equal to visibleLines (or targetLines).length, we know we have finished writing completely. Otherwise, as a DialogBubble is an entity, every single update loop it will add a character to db.currentLine, if that line is finished it will increment db.currentLine.

Reply:
Gotcha, thank you!